src/app_modes/image_view.rs

```diff
@@ -0,0 +1,122 @@
+use self::image::ImageListener;
+use crate::app_modes::{input, AppMode, BaseMode, Drawable};
+use crate::config::ListenerConfig;
+use crate::image;
+use tui::backend::Backend;
+use tui::layout::{Alignment, Constraint, Layout};
+use tui::style::{Color, Style};
+use tui::text::{Span, Spans};
+use tui::widgets::{Block, Borders, Paragraph, Wrap};
+use tui::Frame;
+use tui_image::{ColorMode, Image};
+pub struct ImageView {
+    images: Vec<ImageListener>,
+    active_sub: usize,
+}
+
+impl ImageView {
+    pub fn new(image_topics: Vec<ListenerConfig>) -> ImageView {
+        let mut images: Vec<image::ImageListener> = Vec::new();
+        for image_config in image_topics {
+            images.push(image::ImageListener::new(image_config));
+        }
+        ImageView {
+            images: images,
+            active_sub: 0,
+        }
+    }
+}
+
+impl AppMode for ImageView {
+    fn run(&mut self) {
+        if self.images.len() > 0 && !self.images[self.active_sub].is_active() {
+            self.images[self.active_sub].activate();
+        }
+    }
+
+    fn reset(&mut self) {
+        for sub in self.images.iter_mut() {
+            if sub.is_active() {
+                sub.deactivate();
+            }
+        }
+    }
+
+    fn handle_input(&mut self, input: &String) {
+        if self.images.len() > 0 {
+            match input.as_str() {
+                input::LEFT => {
+                    self.images[self.active_sub].deactivate();
+                    self.active_sub = if self.active_sub > 0 {
+                        self.active_sub - 1
+                    } else {
+                        self.images.len() - 1
+                    };
+                }
+                input::RIGHT => {
+                    self.images[self.active_sub].deactivate();
+                    self.active_sub = (self.active_sub + 1) % self.images.len();
+                }
+                _ => (),
+            }
+        }
+    }
+
+    fn get_description(&self) -> Vec<String> {
+        vec!["This mode allows to visualized images received on the given topics.".to_string()]
+    }
+
+    fn get_keymap(&self) -> Vec<[String; 2]> {
+        vec![
+            [
+                input::LEFT.to_string(),
```

Comment:
i tried to find the key mappings without the help by just smashing keys, here is what i did:

- hit the mode switch to image several times
- hit i
- smash the arrow keys up down
- smash the arrow keys left right
- smash W / S
- smash A / D

Not super intuitive. What do the others say?
-